### PR TITLE
Copy Site: Add copy site to General Settings Site Tools

### PR DESCRIPTION
**Pull request by kozer**
#### Proposed Changes

Add copy site to site tools

#### Testing Instructions

1. Navigate to `Settings -> General -> Site Tools`
2. Ensure you have a site with a `Business` plan
3. Make sure that you see the "Site tools" section

image

4. Press the copy site, and go through the steps. Ensure that your site is copied successfully.
5. Make sure that you can see the analytics tracking when you click the copy site button

image

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Fixes https://github.com/Automattic/dotcom-forge/issues/1508

### CODE REVIEW

File: client/my-sites/site-settings/site-tools/index.jsx [hunk 1]
```diff
@@ -150,35 +176,51 @@ class SiteTools extends Component {
...
+			const siteSlug = getSelectedSiteSlug( state );
+			const isAtomic = isSiteAutomatedTransfer( state, siteId );
+			const planSlug = getSitePlanSlug( state, siteId );
+			const isJetpack = isJetpackSite( state, siteId );
+			const isVip = isVipSite( state, siteId );
+			const isP2 = isSiteWPForTeams( state, siteId );
+			const isP2Hub = isSiteP2Hub( state, siteId );
+			const rewindState = getRewindState( state, siteId );
+			const sitePurchasesLoaded = hasLoadedSitePurchasesFromServer( state );
+			const canSiteBeCopied = siteHasFeature( state, siteId, WPCOM_FEATURES_COPY_SITE ) && planSlug;
```
Comment (danielbachhuber) [inline client/my-sites/site-settings/site-tools/index.jsx]: We need to make sure it's only the site owner, too.

**Review by danielbachhuber — COMMENTED** [review]
Can we create a new hook that exposes two functions, `showCopySiteAction` and `startCopySiteAction` ?

The former should do the permissions checks, and the latter should initiate the copy site process `clearSignupDestinationCookie()`, etc.

We can add the hook on this pull request, and then later update https://github.com/Automattic/wp-calypso/pull/72291 to use it.

**Comment by kozer** [comment]
> The former should do the permissions checks, and the latter should initiate the copy site process `clearSignupDestinationCookie()`, etc.

This is something I was thinking about as well, as we duplicate things here.
However, the hook will not work as this is a class component. I was planning to clean things in the pr you mentioned. Agree.

File: client/my-sites/site-settings/site-tools/index.jsx (hunk 1 above)
Comment (kozer) [reply client/my-sites/site-settings/site-tools/index.jsx]: > We need to make sure it's only the site owner, too.

Wasn't sure about that, thanks for mentioning it. Is it possible to be logged in but not be the site owner, right?

**Comment by danielbachhuber** [comment]
> I was planning to clean things in the pr you mentioned.

@kozer Ok, that works for me too. Can you add the site owner check then, and then we can ship this?

**Review by danielbachhuber — APPROVED** [review]
Looks great, @kozer !

I made a few tweaks to your code ^. I also registered the Tracks event (https://github.com/Automattic/tracks-events-registration/pull/1348) and created an issue for the common abstraction (https://github.com/Automattic/dotcom-forge/issues/1543).

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7766684

Thank you @kozer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7766684).

**Pull request merged**